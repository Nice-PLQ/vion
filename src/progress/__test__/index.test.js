import { mount } from '@vue/test-utils';
import Progress from '../progress.vue';

describe('Progress', () => {
  test('渲染', () => {
    const wrapper = mount(Progress, {
      props: {
        value: 0.5
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('边界判断', async () => {
    const wrapper = mount(Progress, {
      props: {
        value: 1.5
      }
    });
    expect(wrapper.find('.v-progress__text').text()).toBe('100%');
    await wrapper.setProps({ value: -0.5 });
    expect(wrapper.find('.v-progress__text').text()).toBe('0%');
  });

  test('样式', () => {
    const wrapper = mount(Progress, {
      props: {
        value: 0.5,
        showTxt: false,
        background: 'rgb(103, 194, 58)',
        height: '10px'
      }
    });
    expect(wrapper.find('.v-progress__text').exists()).toBe(false);
    expect(wrapper.find('.v-progress__inner').element.style.background).toBe('rgb(103, 194, 58)');
  });

  test('环形进度条', () => {
    const wrapper = mount(Progress, {
      props: {
        value: 0.5,
        type: 'circle',
      }
    });

    // console.log(wrapper.element);
    expect(wrapper.find('.v-progress__circle-text').text()).toBe('50%');
  });
});
