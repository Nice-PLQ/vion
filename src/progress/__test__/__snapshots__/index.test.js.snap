// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress 渲染 1`] = `
<div
  class="v-progress"
>
  
  <div
    class="v-progress__outer"
  >
    <div
      class="v-progress__inner"
      style="width: 50%;"
    />
  </div>
  
  <div
    class="v-progress__text"
  >
    50%
  </div>
  
  
</div>
`;
